src/EnergyPlus/SolarShading.cc

```diff
@@ -4592,6 +4686,22 @@ namespace SolarShading {
 						}
 					}
 				}
+				// Elimate shawdowing surfaces that is supposed to be disabled.
+				if ( DisableAllSelfShading ) {					
+					if ( surface.Zone != 0 ) {
+						continue; // Disable all shadowing surfaces in all zones. Attached shading surfaces are not part of a zone, zone value is 0.
+					}
+				} else if ( DisableGroupSelfShading ) {
+					std::vector< int > DisabledZones = Surface( CurSurf ).DisabledShadowingZoneList;
```

Comment:
I worry about performance here.  We spend a lot of time trying to remedy performance issues already existing in the code, so I am ever more concerned when I see nested loops.

Although I can't see the full context in this diff, at this point in the code, I believe we are inside a loop over each zone, then through each surface.  And now you are looping back over disabled zones again.  It feels passive, and that there is a more intentional, active, approach to this, where you simply alter the shadowing calculations while they are being made, then inside here you never have to loop over them again just to ignore them.

Reply:
But during the shading calculation how it is possible to know which surface is which - non-shading surface, shading surface not to be ignored, shading surface to be ignored?